src/treasury.cairo

```diff
@@ -399,21 +401,21 @@ mod Treasury {
                 token.balanceOf(get_contract_address()) >= transfer_pending.amount,
                 Errors::INSUFFICIENT_FUNDS
             );
-            let status: bool = token.transfer(transfer_pending.receiver, transfer_pending.amount);
+
             let sent_event = TokenSent {
                 amount: transfer_pending.amount,
                 token_addr: transfer_pending.token_addr,
                 receiver: transfer_pending.receiver
             };
-
             self
                 .transfers_on_cooldown
                 .write(
                     transfer_pending.id,
                     Transfer { status: TransferStatus::FINISHED, ..transfer_pending }
                 );
-
             self.emit(sent_event);
+
+            let status: bool = token.transfer(transfer_pending.receiver, transfer_pending.amount);
             status
```

Comment:
token.transfer(transfer_pending.receiver, transfer_pending.amount)

this can be the last line, without the semicolon (;) the return value of the last call becomes the return value of the function.